### PR TITLE
Create Segment CollectionView and PageController

### PR DESCRIPTION
**Pull request by sheep-q**
## [Pull request title]
 - Create SegmentCollectionView to chose Feed Type
 - Create UIPageViewController
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues
#### Screenshots (nếu có)
Screen Shot 2021-04-02 at 3 26 14 PM
Screen Shot 2021-04-02 at 3 26 29 PM

### CODE REVIEW

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 1]
```diff
@@ -8,19 +8,67 @@
...
+            $0.scrollDirection = .horizontal
+            $0.itemSize = CGSize(width: 100, height: 40)
+            $0.sectionInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+            $0.minimumLineSpacing = 1
+            $0.minimumInteritemSpacing = 1
+        }
+        
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.register(SegmentCollectionCell.self,
+                                forCellWithReuseIdentifier:SegmentCollectionCell.identifier)
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift]: dùng lib `Reusable` luôn em

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 2]
```diff
@@ -8,19 +8,67 @@
...
 import UIKit
 
+//MARK: - AppViewController
 class AppViewController: UIViewController {
     
     private let searchBar = UISearchBar()
     
+    private let segmentItems = FeedType.apps.item
+    
+    private var itemPicked : [Bool] = {
```
Comment (dongtg-1485) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift:18]:
```suggestion
    private var itemPicked: [Bool] = {
```

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 3]
```diff
@@ -8,19 +8,67 @@
...
+        var item = [Bool](repeating: false, count: FeedType.apps.item.count)
+        item[0] = true
+        return item
+    }()
+    
+    
+    private let segmentCollection : UICollectionView = {
```
Comment (dongtg-1485) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift:25]:
```suggestion
    private let segmentCollection: UICollectionView = {
```

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 4]
```diff
@@ -67,3 +169,54 @@ extension AppViewController: UISearchBarDelegate {
...
+     
+        let itempick = [Bool](repeating: false, count: FeedType.apps.item.count + 1)
+        self.itemPicked = itempick
+        self.itemPicked[indexPath.row] = true
+        
+        collectionView.reloadData()
+        
+        //push page view controller
+        let nextIndex = indexPath.row
+        let direction = nextIndex < currentPageIndex ? UIPageViewController.NavigationDirection.reverse : UIPageViewController.NavigationDirection.forward
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift]:
```suggestion
        let direction = nextIndex < currentPageIndex 
            ? UIPageViewController.NavigationDirection.reverse 
            : UIPageViewController.NavigationDirection.forward
```

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 5]
```diff
@@ -67,3 +169,54 @@ extension AppViewController: UISearchBarDelegate {
...
+    }
+}
+
+//MARK: - Page ViewController
+extension AppViewController : UIPageViewControllerDelegate, UIPageViewControllerDataSource {
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        currentPageIndex = viewController.view.tag
+        let beforeIndex = viewController.view.tag - 1
+        if beforeIndex < 0 {return nil}
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift]:
```suggestion
        if beforeIndex < 0 { return nil }
```

File: iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift [hunk 6]
```diff
@@ -67,3 +169,54 @@ extension AppViewController: UISearchBarDelegate {
...
+        return childView[beforeIndex]
+    }
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
+        currentPageIndex = viewController.view.tag
+        let afterIndex = viewController.view.tag + 1
+        if afterIndex > childView.count - 1 {return nil}
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/AppViewController/AppViewController.swift]: như trên

File: iOS-Play/iOS-Play/Controller/Tabar Controller/FeedTypeEnum.swift [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        case .apps:
+            return ["New Game","New App","Top Free","Free Ipad","Top Grossing","Grossing Ipad","Top Ipad"]
+        case .musics:
+            return ["hi","hello"]
+        case .movies:
+            return ["hi","hello"]
+        case .books:
+            return ["hi","hello"]
+        case .tvShows:
+            return ["hi","hello"]
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Tabar Controller/FeedTypeEnum.swift]: không để data test như này

File: iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    
+    let bottomView = UIView().then {
+        $0.layer.cornerRadius = 3
+    }
+    
+    let customView = UIView().then {
+        $0.backgroundColor = .white
+    }
+    
+    var title = UILabel().then {
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift]: convention

File: iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        
+        title.snp.makeConstraints { (make) in
+            make.top.right.left.bottom.equalToSuperview()
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift]:
```suggestion
        title.snp.makeConstraints {
            $0.top.right.left.bottom.equalToSuperview()
```

File: iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        
+        customView.snp.makeConstraints { (make) in
+            make.left.right.equalToSuperview()
+            make.bottom.equalToSuperview().inset(-4)
+            make.height.equalTo(3)
+        }
+    }
+
+    public func configure(title: String, color: UIColor, picked: Bool) {
+        self.title.text = title
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift]: bỏ self

File: iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+        if picked {
+            self.bottomView.backgroundColor = .systemPink
+            self.title.textColor = .systemPink
+        } else {
+            self.title.textColor = .darkGray
+            self.bottomView.backgroundColor = color
+        }
```
Comment (datdt-1935) [inline iOS-Play/iOS-Play/Controller/Tabar Controller/SegmentCollectionCell.swift]:
```suggestion
           bottomView.backgroundColor = picked ?  .systemPink : .color
           title.textColor = picked? .systemPink : .darkGray
```

**Review by dongtg-1485 — APPROVED** [review]

**Review by datdt-1935 — APPROVED** [review]

**Review by dongtg-1485 — APPROVED** [review]

**Pull request merged**